### PR TITLE
Recipe App with rails

### PR DESCRIPTION
**Pull request by pinkmoon25**
Created the Recipe app based on the wireframe provided in the [project](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md)

- Add_Models_views_Controllers #11 
  - generate User, Recipe, and Food Models
  - generate Users, Recipes, and Foods controllers
  - generate User, Recipe, and Food Views

- Create_Login_and_Registration_with_Devise #3 
  - Implement login and registration page, built with devise gem.

- Add-Food-Actions-views #4 
  -  Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also 
     quantity of a given food).
  - Should lead to a form that allows users to add new food.

- Add_Recipe_details #5 
  - Should lead to recipe details.
  - If the user is the owner of the recipe, should allow the user to delete

- Add_Recipes_action_views #5 
  - Should lead to recipe details.

- Add Public Recipes #6 
  - Should display a list of all public recipes ordered by newest as in the wireframe.
  - Should lead to recipe details. 

- Add NavBar #10 
  - Create a navigation menu that allows users to open all of the pages you created.
  
- Create Shopping_list #8 
  - Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with 
    the general food list of that user).
  - Should count the total food items and total price of the missing food.
 
- Fix n+1 queries #9 
  - Make sure there are no N+1 queries happening.

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  <div class="field">
+    <%= f.label :password_confirmation, class:'form-label' %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password", class:'form-control' %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up", class:'btn btn-primary mt-3' %>
+  </div>
+<% end %>
```
Comment (prolajumokeoni) [inline app/views/devise/registrations/new.html.erb:32]: - [x] Kindly make sure that a user can sign up successfully, currently, I can not signup successfully to test your application
-  See video below ⬇️ 



https://user-images.githubusercontent.com/69638013/201172657-120d099f-030b-440d-bf89-444b046085b9.mp4

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @PinkMoon25 and @developerwaleed 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors  
✔️  Good PR title and Description



## Required Changes ♻️

- [x] Kindly write  Unit and Integration tests for your application this is part of the technical requirements

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by pinkmoon25** [comment]
Removed email confirmation for easy sign up and Added test cases as requested

**Review by Munsa1 — APPROVED** [review]
Hi @PinkMoon25  @developerwaleed , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**